engine/buildkit/telemetry.go

```diff
@@ -40,36 +42,86 @@ type buildkitTracer struct {
 	provider *buildkitTraceProvider
 }
 
+const TelemetryComponent = "buildkit"
+
 func (t *buildkitTracer) Start(ctx context.Context, spanName string, opts ...trace.SpanStartOption) (context.Context, trace.Span) {
-	opts = append([]trace.SpanStartOption{}, opts...)
+	opts = append([]trace.SpanStartOption{
+		// Sprinkle an attribute on these spans so we can mess with them in the SpanProcessor.
+		//
+		// Ideally Buildkit would just set an appropriate scope name, but it doesn't, so we
+		// have to make do.
+		trace.WithAttributes(attribute.Bool("buildkit", true)),
+	}, opts...)
+
+	ctx, span := t.tracer.Start(ctx, spanName, opts...)
+	newSpan := buildkitSpan{Span: span, provider: t.provider}
+	return trace.ContextWithSpan(ctx, newSpan), newSpan
+}
+
+type buildkitSpan struct {
+	trace.Span
+	provider *buildkitTraceProvider
+}
```

Comment:
Aha, this is *so* much neater now that this logic isn't done entirely in `Start` :tada: